language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -185,7 +185,7 @@ impl<'input> Lexer<'input> {
         len: usize,
     ) -> Result<(), ParseError<usize, anyhow::Error>> {
         self.token = token;
-        self.cur_end = self.cur_start + len;
+        self.cur_end = self.cur_start.wrapping_add(len); // memory will run out long before this wraps
```

Comment:
not sure I follow this, could you expand on what is the expectation?
the `wrapping_add` would silently introduce a bug if I understand this and an overflow panic seems the right thing to do.
Thoughts?

Reply:
The + already silently overflows in non debug builds. It seems unlikely that this will occur in practice since overflowing can only occur in situations that would require a source text so large that memory will have been exhausted before getting to this code. The proposed change just makes this implicit assumption explicit.

Now that it is explicit, you may well decide that you don't like that assumption and instead put a runtime check here in all cases. That is a judgement call. MIRAI will be mollified either way, as long as you intentions and assumptions are explicit.

Reply:
While we are on the subject. There should probably be a fairly restrictive upper limit on the size of a source string. Right now there does not appear to be one.